Netable/Netable/Netable.swift

```diff
@@ -153,117 +182,61 @@ open class Netable {
             headers: urlRequest.allHTTPHeaderFields ?? [:]
         )
 
-        log(.requestStarted(request: requestInfo))
+        await log(.requestStarted(request: requestInfo))
         if !config.enableLogRedaction, let params = try? request.parameters.toParameterDictionary(encodingStrategy: request.jsonKeyEncodingStrategy ?? config.jsonEncodingStrategy) {
-            log(.requestBody(body: params))
+            await log(.requestBody(body: params))
         } else {
             let params = request.unredactedParameters(defaultEncodingStrategy: config.jsonEncodingStrategy)
-            log(.requestBody(body: params))
+            await log(.requestBody(body: params))
         }
 
         let retryConfiguration = self.retryConfiguration
 
-        let task = urlSession.dataTask(with: urlRequest) { data, response, error in
-            let time = CACurrentMediaTime() - startTimestamp
-
+        for retry in 0..<retryConfiguration.count {
             do {
-                if let error = error {
-                    throw NetableError.requestFailed(error)
-                }
-
+                let (data, response) = try await urlSession.data(for: urlRequest)
```

Comment:
This is currently throwing a concurrency warning because in `data(..., delegate: URLSessionTaskDelegate)`, the delegate isn't Sendable but since we're not using it I think it's safe to just ignore?